src/ConfirmProvider.js

```diff
@@ -104,7 +114,7 @@ const ConfirmProvider = ({ children, defaultOptions = {} }) => {
 
   const closeOnParentUnmount = useCallback((parentId) => {
     setState((state) => {
-      if (state && state.parentId === parentId) {
+      if (dialogOptions.closeOnParentUnmount && state && state.parentId === parentId) {
```

Comment:
Hmm, `closeOnParentUnmount` callback is memoized with `useCallback`, so `dialogOptions` here will always be the initial options, which is not what we want.

Technically, there are ways to address it, but they come with tradeoffs, none of which I would like to take.

What is your use case that requires this option?